cogs/music.py

```diff
@@ -0,0 +1,686 @@
+            case "forward_time" | "ft":
+                return (self.forward_time, conv_arg(lambda x: x, True))
+            case "backwards_time" | "bt" | "rewind":
+                return (self.backwards_time, conv_arg(lambda x: "30" if x is None else x, False))
+            case "loop":
+                return (self.loop, conv_arg(lambda x: x, True))
+            case "print_queue" | "pq":
+                return (self.print_queue, conv_arg(lambda x: 0 if x is None else int(x), False))
+            case "status" | "now_playing" | "np":
+                return (self.music_status, None)
+            case "remove" | "pop":
+                return (self.remove_track, conv_arg(int, True))
+            case "insert":
+                return (self.insert_track, conv_arg(lambda x: (lambda y, z: (int(y), z))(*x.split(maxsplit=1)), True))
+            case "clear_queue" | "cq":
+                return (self.clear_queue, None)
+            case "clear_hist" | "ch":
+                return (self.clear_hist, None)
+            case "queue" | "q":
+                return (self.queue_track, lambda x: x)
+            case "volume" | "vol" | "v":
+                return (self.volume, conv_arg(lambda x: float(x.strip("%")), True))
+            case "stop":
+                return (self.stop, None)
+            case "skip" | "next":
+                return (self.skip, conv_arg(lambda x: 0 if x is None else int(x), False))
+            case "back" | "previous":
+                return (self.backtrack, conv_arg(lambda x: None if x is None else int(x), False))
+            case "pause":
+                return (self.pause, None)
+            case "resume":
+                return (self.resume, None)
+            case _:
+                raise ValueError
+
+    @commands.command(aliases=["m"])
+    async def music(self, ctx, subcommand: Optional[str] = None, *, args: Optional[str] = None):
+        """
+        monocommand used for music features.
+        write `?music help subcommand` for details on how each command works (arguments are shown as semicolon separated list).
+        available subcommands are:
+        - play (note: this will interrupt the currently playing song)
+        - playback_speed | ps
+        - goto_time | gt
+        - forward_time | ft
+        - backwards_time | bt | rewind
+        - loop
+        - print_queue | pq
+        - status | now_playing | np
+        - remove | pop
+        - insert
+        - clear_queue | cq
+        - clear_history | ch
+        - queue | q
+        - volume | vol | v
+        - stop
+        - skip | next
+        - back | previous
+        - pause
+        - resume
+        - help
+        """
+
+        if subcommand is None:
+            return await ctx.send(self.music.help)
+        if subcommand == "help":
+            if args is None:
+                return await ctx.send(self.music.help)
+            try:
+                tup = self.subc_decision(args)
```

Comment:
do you ever use tup[1]? if not, consider something like `fn, _ = self.subc...`

Reply:
i don't but i wanna isolate source of error here as much as possible so i think adding the extra unpacking isn't good

not super set on this tho so if you disagree i'll change it